Cypress Component Testing for Angular logs many TypeScript warnings

### Current behavior

When starting Cypress Component Testing in a fresh Angular project, it logs many warnings:

```
WARNING in C:/Sandboxes/Experiments/cypress-component-testing-typescript-warnings/cypress/e2e/spec.cy.ts is part of the TypeScript compilation
 but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

WARNING in C:/Sandboxes/Experiments/cypress-component-testing-typescript-warnings/src/app/app.component.ts is part of the TypeScript compilati
on but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

WARNING in C:/Sandboxes/Experiments/cypress-component-testing-typescript-warnings/src/app/app.module.ts is part of the TypeScript compilation 
but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

WARNING in C:/Sandboxes/Experiments/cypress-component-testing-typescript-warnings/src/main.ts is part of the TypeScript compilation but it's u
nused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.
```

### Desired behavior

Cypress should be configured correctly so that we don't see any warning in the console.

### Test code to reproduce

1.  Create a new Angular project: `ng new cypress-component-testing-typescript-warnings --inline-style --inline-template --minimal --routing false`
2. Run `ng e2e` and add Cypress component testing
3. Run  `cypress open --component --browser chrome`
4. Confirm that several warnings are logged:  
```
WARNING in C:/Sandboxes/Experiments/cypress-component-testing-typescript-warnings/cypress/e2e/spec.cy.ts is part of the TypeScript compilation
 but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

WARNING in C:/Sandboxes/Experiments/cypress-component-testing-typescript-warnings/src/app/app.component.ts is part of the TypeScript compilati
on but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

WARNING in C:/Sandboxes/Experiments/cypress-component-testing-typescript-warnings/src/app/app.module.ts is part of the TypeScript compilation 
but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

WARNING in C:/Sandboxes/Experiments/cypress-component-testing-typescript-warnings/src/main.ts is part of the TypeScript compilation but it's u
nused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.
```


### Cypress Version

12.9.0

### Node version

18.14.2

### Operating System

Win10

### Debug Logs

_No response_

### Other

_No response_

I had the same issue on a large app (~10000 ts files) and cypress takes some minutes to launch a single test with so much warning about unused compiled code.

I think cypress launch an angular debug server for component testing and use the tsconfig.app.json specified in angular.json in build or serve part.

It can be helpful if cypress can use a specific tsconfig by convention like with E2E testing

I have done a quick reverse engineering, so if i rename tsconfig.app.json, i had this stacktrace at launch : 
```
Error: error TS5012: Cannot read file '/home/me/projects/web/tsconfig.app.json': ENOENT: no such file or directory, open '/home/me/projects/web/tsconfig.app.json'.
    at readTsconfig (/home/packages/angular_devkit/build_angular/src/utils/read-tsconfig.ts:35:11)
    at async generateWebpackConfig (/home/packages/angular_devkit/build_angular/src/utils/webpack-browser-config.ts:43:20)
    at async generateBrowserWebpackConfigFromContext (/home/packages/angular_devkit/build_angular/src/utils/webpack-browser-config.ts:155:18)
    at async getAngularCliWebpackConfig (/home/me/.cache/Cypress/12.8.1/Cypress/resources/app/node_modules/@packages/server/node_modules/@cypress/webpack-dev-server/dist/helpers/angularHandler.js:145:24)
    at async angularHandler (/home/me/.cache/Cypress/12.8.1/Cypress/resources/app/node_modules/@packages/server/node_modules/@cypress/webpack-dev-server/dist/helpers/angularHandler.js:180:27)
    at async getPreset (/home/me/.cache/Cypress/12.8.1/Cypress/resources/app/node_modules/@packages/server/node_modules/@cypress/webpack-dev-server/dist/devServer.js:91:20)
    at async Function.devServer.create (/home/me/.cache/Cypress/12.8.1/Cypress/resources/app/node_modules/@packages/server/node_modules/@cypress/webpack-dev-server/dist/devServer.js:108:61)
    at async /home/me/.cache/Cypress/12.8.1/Cypress/resources/app/node_modules/@packages/server/node_modules/@cypress/webpack-dev-server/dist/devServer.js:25:24
 ELIFECYCLE  Command failed with exit code 1.

--- a/cli/CHANGELOG.md
+++ b/cli/CHANGELOG.md
@@ -37,6 +37,7 @@ _Released 08/12/2025 (PENDING)_
 - Upgraded `tmp` from `~0.2.3` to `~0.2.4`. This removes the [CVE-2025-54798](https://github.com/advisories/GHSA-52f5-9888-hmc6) vulnerability being reported in security scans. Addresses [#32176](https://github.com/cypress-io/cypress/issues/32176).
 - Fixed an issue where `.fixture()` calls with a specified encoding would sometimes still attempt to parse the file based on its extension. Files with an explicit encoding are now always treated as raw content. Fixes [#32139](https://github.com/cypress-io/cypress/issues/32139).
 - Fixed an issue where `.fixture()` calls with different encoding options would return inconsistent content based on execution order. Fixes [#32138](https://github.com/cypress-io/cypress/issues/32138).
+- Fixed an issue where Angular Component Testing was printing extraneous warnings to the console by default. By default, errors only will now print to the console. This can still be overridden by passing in a custom webpack config or setting the `verbose` option inside your `angular.json`. Addresses [#26456](https://github.com/cypress-io/cypress/issues/26456).
 
 **Misc:**
 

--- a/npm/webpack-dev-server/src/helpers/angularHandler.ts
+++ b/npm/webpack-dev-server/src/helpers/angularHandler.ts
@@ -298,6 +298,14 @@ async function getAngularCliWebpackConfig (devServerConfig: AngularWebpackDevSer
     },
   )
 
+  // @see https://github.com/cypress-io/cypress/issues/26456
+  // if verbose is not set, we set the stats to errors-only
+  // users can either override this stat by passing in a custom webpack config
+  // or by setting the verbose option to true
+  if (!buildOptions.verbose) {
+    config.stats = 'errors-only'
+  }
+
   delete config.entry.main
 
   return config

--- a/npm/webpack-dev-server/test/handlers/angularHandler.spec.ts
+++ b/npm/webpack-dev-server/test/handlers/angularHandler.spec.ts
@@ -38,6 +38,7 @@ describe('angularHandler', function () {
     expect(webpackConfig).to.exist
     expect((webpackConfig?.entry as any).main).to.be.undefined
     expect(sourceWebpackModulesResult.framework?.importPath).to.include(path.join('@angular-devkit', 'build-angular'))
+    expect(webpackConfig.stats).to.equal('errors-only')
     const projectConfig = await getProjectConfig(projectRoot)
 
     expect(projectConfig).to.deep.eq({

